lib/acme/certificate.rb

```diff
@@ -9,6 +9,39 @@ def initialize(certificate, chain, request)
     @x509 = certificate
     @x509_chain = chain
     @request = request
+
+    #create_self_signed_cert if x509.nil?
+  end
+
+  def create_self_signed_cert subjectAltDomains
+    keypair = OpenSSL::PKey::RSA.new(2048) 
+
+    cert = OpenSSL::X509::Certificate.new
+    cert.subject = cert.issuer = OpenSSL::X509::Name.parse "/CN=#{subjectAltDomains.first}"
+    cert.not_before = Time.now
```

Comment:
Is setting this explicitly really necessary? Besides that it seems to be good practice to date back a certificate a little bit (an hour or so) to accommodate for clock drift, DST changes etc.